classes/Http/Action/Profile/EditAction.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Http\Action\Profile;
+
+use OpenCFP\Domain\Model;
+use OpenCFP\Domain\Services;
+use OpenCFP\PathInterface;
+use Symfony\Component\HttpFoundation;
+use Symfony\Component\Routing;
+use Twig_Environment;
+
+final class EditAction
+{
+    /**
+     * @var Services\Authentication
+     */
+    private $authentication;
+
+    /**
+     * @var PathInterface
+     */
+    private $path;
+
+    /**
+     * @var Twig_Environment
+     */
+    private $twig;
+
+    /**
+     * @var Routing\Generator\UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    public function __construct(
+        Services\Authentication $authentication,
+        PathInterface $path,
+        Twig_Environment $twig,
+        Routing\Generator\UrlGeneratorInterface $urlGenerator
+    ) {
+        $this->authentication = $authentication;
+        $this->path           = $path;
+        $this->twig           = $twig;
+        $this->urlGenerator   = $urlGenerator;
+    }
+
+    public function __invoke(HttpFoundation\Request $request): HttpFoundation\Response
+    {
+        $user = $this->authentication->user();
+
+        if ((string) $user->getId() !== $request->get('id')) {
+            $request->getSession()->set('flash', [
+                'type'  => 'error',
+                'short' => 'Error',
+                'ext'   => "You cannot edit someone else's profile",
+            ]);
+
+            $url = $this->urlGenerator->generate('dashboard');
+
+            return new HttpFoundation\RedirectResponse($url);
+        }
+
+        $speakerData = Model\User::find($user->getId())->toArray();
```

Comment:
Because of static access here we cannot write unit tests that go further than the condition above.

Reply:
I've been thinking about this sort of thing for a while now. It occurs to me that monkey patching is the best solution here, but I don't know if there is a good solution that works with PHPUnit.